src/rust/engine/fs/src/directory.rs

```diff
@@ -958,7 +958,7 @@ impl DigestTrie {
 
             if let Some(Entry::Symlink(s)) = maybe_matching_entry {
                 if link_depth >= MAX_LINK_DEPTH {
-                    warn!("Exceeded the maximum link depth while traversing links. Stopping traversal.");
+                    warn!("Exceeded the maximum link depth while traversing link `{}` to path {:#?}. Stopping traversal.", s.name, s.target);
```

Comment:
```suggestion
                    warn!("Exceeded the maximum link depth while traversing link {:#?} to path {:#?}. Stopping traversal.", path_so_far, s.target);
```

Reply:
This one is cleaner because we can just use the `logical_path` joined with `s.name`.